java/java.platform.ui/src/org/netbeans/api/java/platform/PlatformComboBoxModel.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.api.java.platform;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.AbstractListModel;
+import javax.swing.ComboBoxModel;
+import org.openide.util.WeakListeners;
+
+/**
+ * A {@link ComboBoxModel} holding {@link JavaPlatform} references. The selected
+ * object can be a JavaPlatform or any object whose toString() method returns
+ * a registered Java Platform ID. This platform id can be obtained from a
+ * Java platform by getting its {@link #PROP_PLATFORM_ID} property.
+ * <pre>
+ *     JavaPlatform platform = &lt;some java platform&gt;
+ *     String id = platform.getProperties().get(PROP_PLATFORM_ID);
+ *     ...
+ *     PlatformComboBoxModel model = new PlatformComboBoxModel();
+ *     model.setSelectedItem(id);
+ * </pre>
+ *
+ * @since 1.51
+ * @author Laszlo Kishalmi
+ */
+public final class PlatformComboBoxModel extends AbstractListModel<JavaPlatform> implements ComboBoxModel<JavaPlatform>, PropertyChangeListener {
```

Comment:
I suggest to hide this class behind factory method.

Reply:
The updated PR does that. Good idea. Thank you!